### DISCUSSION TITLE
Best solution for .env.dev & .env.prod

### DISCUSSION THREAD

**Discussion by sellrock-de** [dunglas/symfony-docker#306] [General]
Is this the best solution to do 

`ADD .env.${APP_ENV} /srv/app/.env `

in Dockerfile?

**Comment by maxhelias** [comment]
The current configuration of the project is the right solution. As reminder, these files must not contain any information sensitive to your environment because there are committed. Use `.env.local` or `.env.<environment>.local` for override the default values. See https://symfony.com/doc/current/configuration.html#overriding-environment-values-via-env-local

**Reply by ousmaneNdiaye** [reply]
@maxhelias do you mean that setting vars in `.env.local` should work out of the box (like in a regular Symfony project without Docker) ?

Because my Docker container seems to always expose env vars from `.env`... 
I don't know what I'm missing... 🧐 

Is any command to run before docker build in order to user .env.local ?
Thanks

**Reply by aaa2000** [reply]
I think you should use `env_file` in  docker-compose.yml

```
services:
  php:
    env_file:
      - ./.env
      - ./.env.local
...
```

**Reply by maxhelias** [reply]
Symfony like docker uses the .env file to expose the env vars.
The file `.env.local` or `.env.<environment>.local` are exposed on symfony runtime.
Read the full article above to learn more about best practices.

**Reply by ousmaneNdiaye** [reply]
Yes I am familiar with the Symfony env var mechanism, but unfortunately my container always seems to use `.env` file 🤷‍♂️

**Reply by maxhelias** [reply]
Your `.env` are always exposed to symfony and docker
